src/Nethermind/Nethermind.State/PersistentStorageProvider.cs

```diff
@@ -213,6 +207,7 @@ protected override void CommitCore(IStorageTracer tracer)
             if (_stateProvider.AccountExists(address))
             {
                 _toUpdateRoots.Add(address);
+                GetOrCreateStorage(address, out _);
```

Comment:
Why is it needed to create one? Comment please or fix.

Reply:
It can't create them on demand in the parallel loop as the storage dictionary is not concurrent

Reply:
Added comment